pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerStatsRecorderImpl.java

```diff
@@ -218,44 +224,63 @@ void updateCumulativeStats(ProducerStats stats) {
         totalBytesSent.add(stats.getNumBytesSent());
         totalSendFailed.add(stats.getNumSendFailed());
         totalAcksReceived.add(stats.getNumAcksReceived());
+        sendMsgsRate += stats.getSendMsgsRate();
+        sendBytesRate += stats.getSendBytesRate();
+        if (latencyPctValues == null || latencyPctValues.length == 0) {
+            latencyPctValues = stats.getLatencyPctValues();
+        } else {
+            double[] curLatencyPctValues = stats.getLatencyPctValues();
+            for (int i = 0; i < this.latencyPctValues.length; i++) {
+                this.latencyPctValues[i] += curLatencyPctValues[i];
+            }
+        }
     }
 
     @Override
     public long getNumMsgsSent() {
-        return numMsgsSent.longValue();
+        return currentNumMsgsSent;
```

Comment:
Seems this is changing the original behavior here? 
before this change, we will get value in numMsgsSent, which is changing according to messages sent; 
after this change, it is a static value, which is inited in method `init()` at each interval.

Reply:
Yes, but I think `numMsgsSent` should be consistent with `sendMsgsRate` as a stat metric, generated as a collection of every timer task.
The previous meaning is based on the sending of the message, and several aggregated values in the timer task may cause inconsistency.

e.g. `numMsgsSent`: 10 `sendMsgsRate`: 0 (when stat timing task has not been counted)
or  `numMsgsSent`: 0 `sendMsgsRate`: 8.5 (No message is sent after numMsgsSent.sumThenReset() is called, the timer task has not been executed again)

Reply:
Different calibers may confuse users

Reply:
make sense to me. 
@merlimat Would you please help confirm this? I am not sure if it is intentional to do this.